app/controllers/application_controller/timelines.rb

```diff
@@ -93,10 +93,11 @@ def tl_build_timeline_report_options
       from_dt = create_time_in_utc("#{from_date.strftime} 00:00:00",
                                    session[:user_tz])
 
-      rec_cond, *rec_params = @tl_record.event_where_clause(@tl_options.evt_type)
-      conditions = [rec_cond, "timestamp >= ?", "timestamp <= ?"]
+      rec_cond, *rec_params = @tl_record.event_where_clause(@tl_options.evt_type, params[:tl_storage_systems])
```

Comment:
So the comment seems to have been lost in the changes, but this is still an issue.  We can't be adding in provider specifics like storage_systems here in the general timeline options.

I think it is better to have `@tl_record` be your EMS and you can return all the storages related to the ems (this also fixes your RBAC issue) and/or have `@tl_record` be a specific storage system (also resolves the RBAC issue)

Reply:
The suggested change is the situation today (Master branch), where when using the EMS, all the physical storages are displayed, and when selecting specific storage only its events are presented.

The params are returned from the form the user filled, so not passing them will mean no filtering, which is the purpose of the PR.

As I see it, the change will have to add provider-specific code somewhere in the code (which I agree is not good), so the problem will be moved to another file or function. Is there a solution that you think might fit here?

Reply:
@jeffibm @GilbertCherrie @DavidResende0 is there a way for the Autosde provider to override the timeline UI page without modifying the core options?